### PR TITLE
web: add http.status_code to span data

### PR DESCRIPTION
**Pull request by tatchi**
This adds `http.status_code` to the span data

### CODE REVIEW

**Comment by tatchi** [comment]
cc @ELLIOTTCABLE

File: web.ml [hunk 1]
```diff
@@ -316,6 +316,7 @@ module Http (IO : IO_TYPE) (Curl_IO : CURL with type 'a t = 'a IO.t) : HTTP with
     let t = new Action.timer in
     let result = Some (fun h code ->
       if verbose then verbose_curl_result nr_http action t h code;
+      Possibly_otel.add_data_to_manual_span sid ["http.status_code", `Int (Curl.get_httpcode h)];
```
Comment (Khady) [inline web.ml:319]: why not calling Trace_core.add_data_to_manual_span given that it's not a function that is being stubbed?

**Comment by Khady** [comment]
it looks good to me but I don't know how to check that it does the right thing

**Comment by Khady** [comment]
should it be named `http.response.status_code`? https://opentelemetry.io/docs/specs/semconv/attributes-registry/http/#http-attributes

**Comment by Khady** [comment]
validated internally that it works. So can merge this version and change to a different name as necessary (or log with both names?)

**Review by Khady — APPROVED** [review]
thanks

**Pull request merged**